### ISSUE TITLE
Time and Duration support?

### ISSUE THREAD

**Issue by natefinch** [google/skylark#17]
I know bazel's skylark eschews time functions because they want to ensure reproducibility in their configurations, but this implementation of skylark seems like it could be used for so much more, and at least having access to official skylark time and duration types would keep implementations compatible that otherwise would not be, if consumers all create their own implementations.

Even purely from a configuration standpoint, I could definitely see wanting to be able to specify durations for timeouts, or timestamps for expiration dates, etc.

**Comment by alandonovan** [comment]
Yes, if Skylark were to have a standard library, time & date would be an obvious first package.  For now I suggest you experiment with the package interface you want, and later we can discuss how to share it with others.

**Comment by natefinch** [comment]
cool. 

FYI, I have a helper wrapper I'm working on here: https://godoc.org/github.com/hippogryph/skyhook

It's mostly just a convenience wrapper using reflection to avoid boilerplate. 

I'm happy to make a helper library for use with time values.

**Comment by natefinch** [comment]
time implementation coming along here: https://github.com/hippogryph/time

**Comment by natefinch** [comment]
So, a question on APIs ... are we going for the same API as python, i.e. everything should work as in https://docs.python.org/2/library/datetime.html transparently?

**Comment by alandonovan** [comment]
Don't copy Python---they've got it wrong so many times.  I would use the Go API as a starting point, something like this:

```
module time

functions
        duration(string) duration                               # parse a duration
        location(string) location                               # parse a location
        time(string, format=..., location=...) time             # parse a time
        now() time # this can't be a standard feature though
        zero time # a constant

type duration
operators
        duration - time = duration
        duration + time = time
        duration == duration
        duration < duration
fields
        hours float
        minutes float
        nanoseconds int
        seconds float

type time
operators
        time == time
        time < time  
        time + duration = time
        time - duration = time
        time - time = duration
fields
        year int
        month int
        day int
        hour int
        minute int
        second int
        nanosecond int

TODO:
- unix time_t conversions
- timezone stuff
- strftime formatting
- constructor from 6 components + location
```
